### PR TITLE
Do not remove typehints inside a method definition

### PR DESCRIPTION
**Pull request by sroze**
This PR fixes #814 by simply not removing typehints found inside a method definition.

### CODE REVIEW

**Comment by stof** [comment]
Tests are broken on PHP 7

**Comment by sroze** [comment]
@stof yop yop I'll have a look today

**Comment by sroze** [comment]
@ciaranmcnulty :tada:

File: src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php [hunk 1]
```diff
@@ -99,13 +99,34 @@ private function stripTypeHints($tokens)
...
                     }
                     break;
+                case self::STATE_READING_FUNCTION_BODY:
+                    if ('{' == $token) {
+                        $this->currentBodyLevel++;
+                    }
+                    elseif ('}' == $token) {
+                        $this->currentBodyLevel--;
+
+                        if ($this->currentBodyLevel == 0) {
```
Comment (stof) [inline src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php]: this should use a strict comparison

File: src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php [hunk 2]
```diff
@@ -99,13 +99,34 @@ private function stripTypeHints($tokens)
...
+                            $this->currentFunction = '';
+                        }
+                    }
+
+                    if (!$this->currentFunction) {
+                        $this->state = self::STATE_READING_CLASS;
```
Comment (stof) [inline src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php]: this state change should happen in the ``$this->currentBodyLevel == 0`` condition IMO. This is where you know you are leaving the function

**Comment by sroze** [comment]
Makes sense, addressed your points @stof.

File: spec/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriterSpec.php [hunk 3]
```diff
@@ -70,6 +70,39 @@ public function bar($bar)
         ');
     }
 
+    function it_do_not_remove_typehints_in_methods()
```
Comment (jon-acker) [inline spec/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriterSpec.php]: do -> does

Comment (ciaranmcnulty) [inline spec/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriterSpec.php]: Just this to change from what I can see

**Comment by sroze** [comment]
@ciaranmcnulty I fixed the typo reported by @jon-acker :)

**Comment by ciaranmcnulty** [comment]
Thanks, @sroze!

**Pull request merged**